retrigger/triggerhandler.py

```diff
@@ -386,6 +388,24 @@ async def check_triggers(self, message: discord.Message, edit: bool) -> None:
                 await self.perform_trigger(message, trigger, search[1])
                 return
 
+    def convert_embed_to_string(self, embed: discord.Embed) -> str:
+        embed_dict = embed.to_dict()
+        flattened_embed_dict = {}
+        field_blacklist = ["type", "timestamp", "color", "proxy_url", "height", "width", "proxy_icon_url"]
+        for field, value in {k: v for k, v in embed_dict.items() if k not in field_blacklist}.items():
+            if type(value) is dict:
+                for subfield in {k for k in value if k not in field_blacklist}:
+                    flattened_embed_dict[field.capitalize() + subfield.capitalize()] = value[subfield]
+            elif type(value) is list:
+                for embedfields in value:
+                    flattened_embed_dict[field.capitalize() + embedfields["name"].capitalize()] = embedfields["value"]
+            else:
+                flattened_embed_dict[field.capitalize()] = value
+        embed_string = ''
+        for field, value in flattened_embed_dict.items():
+            embed_string += f"\n{field}: {value}"
+        return embed_string
+
```

Comment:
```suggestion
    @staticmethod
    def convert_embed_to_string(embed: discord.Embed, embed_index: int = 0) -> str:
        embed_dict = embed.to_dict()
        flattened_embed_dict = {}
        field_blacklist = ["type", "color", "proxy_url", "height", "width", "proxy_icon_url"]
        for field, value in embed_dict.items():
            if field in field_blacklist:
                continue
            if isinstance(value, dict):
                for subfield in value:
                    if subfield in field_blacklist:
                        continue
                    flattened_embed_dict[f"{field.lower()}-{subfield.lower()}"] = value[subfield]
            elif isinstance(value, list):
                for embedfields in value:
                    emfield_name = embedfields["name"]
                    flattened_embed_dict[f"{field.lower()}-{emfield_name}"] = embedfields["value"]
            else:
                flattened_embed_dict[field.lower()] = value
        return "\n".join(f"embed-{embed_index}-{field}: {value}" for field, value in flattened_embed_dict.items())

```
Some minor changes here, I feel like the string should designate that it is an embed and what index the embed is at for things like twitter links that technically contain multiple embeds in order to display multiple images in the embed. 

This should also be a staticmethod since we don't need anything from its parent class passed into it. 

In order to improve regex parsing for end users making the names lowercase and separate extra info by `-` to make it easier to define what part of the embed a user may want to search. e.g. `embed-\d-footer-text:(.+)(?!\n)` would capture only the content of the embeds footer in group 1. 

I also reduced the number of iterations, the inline dictionary manipulation would technically be iterating over each dict twice when we can just ignore the blacklist during the first iteration.

I re-added the timestamp because I have some cool ideas for parsing the timestamp and potentially converting to a discord timestamp, not sure if anything will pan out from that but ISO timestamps look pretty straightforward to build regex for and could be useful.

This should also technically be added in `abc.py` as an abstract staticmethod but I can add that later.